src/main/scala/viper/gobra/frontend/info/implementation/typing/ghost/separation/GhostAssignability.scala

```diff
@@ -230,6 +233,33 @@ trait GhostAssignability {
     }
   }
 
+  /** returns true if `call` should be treated as being ghost because it is not explicitly
+    * annotated as being ghost _and_ one of the arguments is ghost even though the corresponding
+    * parameter is not explicitly annotated as being ghost.
+    */
+  private def isCallImplicitlyGhost(call: ap.FunctionCall): Boolean = {
+    val isPure = call.callee match {
+      case p: ap.Function => p.symb.isPure
+      case p: ap.Closure => p.symb.isPure
+      case p: ap.MethodExpr => p.symb.isPure
+      case p: ap.ReceivedMethod => p.symb.isPure
+      case p: ap.BuiltInFunction => p.symb.isPure
+      case p: ap.BuiltInMethodExpr => p.symb.isPure
+      case p: ap.BuiltInReceivedMethod => p.symb.isPure
+      case p: ap.ImplicitlyReceivedInterfaceMethod => p.symb.isPure
+      case _: ap.DomainFunction => true
+      case _ => false
+    }
```

Comment:
same as before

Reply:
is this comment a duplicate of the comment about lines 242-251 or are they about different things? I've changed the trait such that it's `call.callee.isPure` now

Reply:
oops, my bad, this is a duplicate